azext_edge/edge/providers/check/base.py

```diff
@@ -690,3 +710,8 @@ def resources_grouped_by_namespace(resources: List[dict]):
 
 def filter_by_namespace(resources: List[dict], namespace: str) -> List[dict]:
     return [resource for resource in resources if get_resource_namespace(resource) == namespace]
+
+
+def generate_target_resource_name(api_info: EdgeResourceApi, resource_kind: str) -> str:
+    resource_plural = api_info._kinds[resource_kind] if api_info._kinds else f"{resource_kind}s"
```

Comment:
Right now, we have to do api_info.kinds or api_info.get_resources to populate api_info._kinds dictionary. We will improve.

Reply:
That prior mentioned pre-req is being called earlier in the flow right?

Reply:
no, thanks